kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/jbpm/cases/CaseServiceIntegrationTest.java

```diff
@@ -0,0 +1,245 @@
+            assertEquals(2, caseData.size());
+            assertEquals("fiat", caseData.get("car"));
+
+            Object caseClaimReport = caseData.get("claimReport");
+            assertNotNull(caseClaimReport);
+            assertEquals(caseClaimReport.getClass().getName(), CLAIM_REPORT_CLASS_NAME);
+
+            caseClient.removeCaseInstanceData(CONTAINER_ID, caseId, "claimReport");
+            caseData = caseClient.getCaseInstanceData(CONTAINER_ID, caseId);
+            assertNotNull(caseData);
+            assertEquals(1, caseData.size());
+            assertEquals("fiat", caseData.get("car"));
+
+            Map<String, Object> data = new HashMap<>();
+            data.put("owner", "john");
+            data.put("report", caseClaimReport);
+            caseClient.putCaseInstanceData(CONTAINER_ID, caseId, data);
+
+            caseData = caseClient.getCaseInstanceData(CONTAINER_ID, caseId);
+            assertNotNull(caseData);
+            assertEquals(3, caseData.size());
+            assertEquals("fiat", caseData.get("car"));
+            assertEquals("john", caseData.get("owner"));
+
+            caseClaimReport = caseData.get("report");
+            assertNotNull(caseClaimReport);
+            assertEquals(caseClaimReport.getClass().getName(), CLAIM_REPORT_CLASS_NAME);
+        } catch (Exception e) {
+            fail("Failed due to " + e.getMessage());
+        } finally {
+            if (caseId != null) {
+                caseClient.cancelCaseInstance(CONTAINER_ID, caseId);
+            }
+        }
+    }
+
+    @Test
+    public void testCreateCaseWithEmptyCaseFileThenDestroyIt() {
+        String caseId = caseClient.startCase(CONTAINER_ID, CASE_DEF_ID);
+        try {
+            assertNotNull(caseId);
+            assertTrue(caseId.startsWith(CASE_ID_PREFIX));
+
+            CaseInstance caseInstance = caseClient.getCaseInstance(CONTAINER_ID, caseId);
+            assertNotNull(caseInstance);
+            assertEquals(caseId, caseInstance.getCaseId());
+            assertEquals(CASE_DEF_ID, caseInstance.getCaseDefinitionId());
+            assertEquals(CASE_DESCRIPTION, caseInstance.getCaseDescription());
+            assertEquals(USER_YODA, caseInstance.getCaseOwner());
+            assertEquals(ProcessInstance.STATE_ACTIVE, caseInstance.getCaseStatus().intValue());
+            assertNotNull(caseInstance.getStartedAt());
+            assertNull(caseInstance.getCompletedAt());
+            assertEquals("", caseInstance.getCompletionMessage());
+
+            // since roles were not assigned to any users/groups no tasks are available
+            List<TaskSummary> tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertEquals(0, tasks.size());
+
+            caseClient.destroyCaseInstance(CONTAINER_ID, caseId);
+
+            try {
+                // this should throw exception as there is no case any more
+                caseClient.getCaseInstance(CONTAINER_ID, caseId);
```

Comment:
Adding a fail() in the next line would ensure that the exception was actually thrown in the test.